src/routes/add-article.ts

```diff
@@ -1,16 +1,24 @@
 import createHttpError from 'http-errors';
 import { constants } from 'http2';
-import { JsonLdObj } from 'jsonld/jsonld-spec';
+import jsonld from 'jsonld';
+import { JsonLdArray } from 'jsonld/jsonld-spec';
 import { Next } from 'koa';
 import uniqueString from 'unique-string';
+import { schema } from 'rdf-namespaces';
 import { AppContext, AppMiddleware } from '../app';
 
 export default (): AppMiddleware => (
   async ({ articles, request, response }: AppContext, next: Next): Promise<void> => {
-    const article = request.body as JsonLdObj;
+    const [article] = await jsonld.expand(request.body) as JsonLdArray;
+    if (!(article['@type'].includes(schema.Article))) {
+      throw new createHttpError.Forbidden(`Article type must be ${schema.Article} ('${article['@type'].join(', ')}' was given)`);
+    }
```

Comment:
This should probably be on the article repository.

Reply:
This check has been moved to `in-memory-articles`. The error thrown there leads to `400 Bad Request`.